File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int n : nums){
            map.put(n, map.getOrDefault(n, 0) + 1);
        }
        
        for(int key : map.keySet()){
            System.out.println(key + ":" + map.get(key));
        }
        
        int missing = 0;
        for(int i = 0; i < nums.length + 1; i++){
            
            if(!map.containsKey(i)){
                missing = i;
            }
            // System.out.println(i);
        }
        
        return missing;
        
    }
}

File: 0268-missing-number/0268-missing-number.py
# class Solution:
#     def missingNumber(self, nums: List[int]) -> int:
#         lookup = Counter(nums)
#         missing = 0
#         for i in range(1, len(nums) + 1):
#             if i not in lookup:
#                 missing = i

#         return missing
    

    
class Solution:
    def missingNumber(self, nums):
        mp = {n: i for i, n in enumerate(nums)}
            
        for i in range(len(nums)+1):
            if i not in mp:
                return i
        return 0